src/diffusers/models/modeling_flax_utils.py

```diff
@@ -435,6 +435,7 @@ def from_pretrained(
             # make sure all arrays are stored as jnp.ndarray
             # NOTE: This is to prevent a bug this will be fixed in Flax >= v0.3.4:
             # https://github.com/google/flax/issues/1261
+        state = cls._cast_floating_to(state, dtype)
```

Comment:
@SauravMaheshkar did you forget to update `_cast_floating_to`? I don't think this works as it's not a class method in the current implementation.